src/main/java/com/flowpowered/noise/module/modifier/Range.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * Copyright (c) 2013 Flow Powered <https://flowpowered.com/>
+ * Original libnoise C++ library by Jason Bevins <http://libnoise.sourceforge.net/>
+ * jlibnoise Java port by Garrett Fleenor <https://github.com/RoyAwesome/jlibnoise>
+ * Flow Noise is re-licensed with permission from jlibnoise author.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.flowpowered.noise.module.modifier;
+
+import com.flowpowered.noise.exception.NoModuleException;
+import com.flowpowered.noise.module.Module;
+
+/**
+ * A modifier module to map a value from one range to another.
+ * For example: 0.5 (-1 &lt; x &lt; 1) -&gt; 0.75 (0 &lt; x &lt; 1)
+ */
+public class Range extends Module {
+    
+    public static final double DEFAULT_CURRENT_LOWER_BOUND = -1f;
+    public static final double DEFAULT_CURRENT_UPPER_BOUND = 1f;
+    public static final double DEFAULT_NEW_LOWER_BOUND = 0f;
+    public static final double DEFAULT_NEW_UPPER_BOUND = 1f;
+    
+    /* Current lower bound */
+    private double currentLowerBound = DEFAULT_CURRENT_LOWER_BOUND;
+    /* Current upper bound */
+    private double currentUpperBound = DEFAULT_CURRENT_UPPER_BOUND;
+    /* New lower bound */
+    private double newLowerBound = DEFAULT_NEW_LOWER_BOUND;
+    /* New Upper Bound */
+    private double newUpperBound = DEFAULT_NEW_UPPER_BOUND;
+    /* Cache variables */
+    private double scale;
+    private double bias;
+
+    public Range() {
+        super(1);
+    }
+
+    public double getCurrentLowerBound() {
+        return currentLowerBound;
+    }
+
+    // Bounds setters are private to prevent changes to bounds
+    // without recalculating scale/bias
+    private void setCurrentLowerBound(double currentLowerBound) {
+        this.currentLowerBound = currentLowerBound;
+    }
+
+    public double getCurrentUpperBound() {
+        return currentUpperBound;
+    }
+
+    private void setCurrentUpperBound(double currentUpperBound) {
```

Comment:
The setters should all be public. It doesn't matter if the recalculate method is called a few times, it should only be used once it the lifetime of the module. The lifetime is normally that of the entire app, so it's trivial overhead.